From 961d1db17ff95313d295713cb299b4b1cfdb1f54 Mon Sep 17 00:00:00 2001
From: MagneFire <IDaNLContact@gmail.com>
Date: Sat, 13 Feb 2021 17:01:29 +0100
Subject: [PATCH] mdss: Import video driver from Marshmallow. For some reason a
 garbled output is observed when using Nougat based kernels. Using the
 Marshmallow implementation fixes this.

---
 drivers/video/msm/mdss/mdp3.c                 |  66 +-

diff --git a/drivers/video/msm/mdss/mdp3.c b/drivers/video/msm/mdss/mdp3.c
index 052c9964425..0b131abe9d4 100644
--- a/drivers/video/msm/mdss/mdp3.c
+++ b/drivers/video/msm/mdss/mdp3.c
@@ -2209,11 +2194,11 @@ static int mdp3_is_display_on(struct mdss_panel_data *pdata)
 	}
 
 	mdp3_res->splash_mem_addr = MDP3_REG_READ(MDP3_REG_DMA_P_IBUF_ADDR);
-	
-	if ((pdata->panel_info.type == MIPI_CMD_PANEL) ||(!rc)) {
-		if (mdp3_clk_enable(0, 0))
-			pr_err("fail to turn off MDP core clks\n");
-	}
+
+	if (pdata->panel_info.type == MIPI_CMD_PANEL)
+	rc = mdp3_clk_enable(0, 0);
+	if (rc)
+		pr_err("fail to turn off MDP core clks\n");
 	return rc;
 }
 
-- 
2.30.1

